py3status/modules/vpn_status.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+"""
+Drop-in replacement for i3status run_watch VPN module.
+
+Expands on the i3status module by displaying the name of the connected vpn
+using the NetworkManager python module.
+
+Configuration parameters:
+    cache_timeout: How often we refresh this module in seconds.
+        (default 10)
+    check_pid: If true, act just like the default i3status module.
+        (default False)
+    format: Format of the output.
+        (default 'VPN: {name}')
+    pidfile: Same as i3status.conf pidfile, checked when check_pid is True.
+        (default '/sys/class/net/vpn0/dev_id')
+
+Format string parameters:
+    {name} The name and/or status of the VPN.
+
+Requires:
+    python-NetworkManager
+
+@author Nathan Smith <nathan AT praisetopia.org>
+"""
+
+try:
+    import NetworkManager
+except ImportError:
+    fallback = True
+else:
+    import dbus
+    from dbus.mainloop.glib import DBusGMainLoop
+
+    DBusGMainLoop(set_as_default=True)
+    from gi.repository import GLib
+
+    fallback = False
+from threading import Thread
+from os import path
+from time import time
+
+
+class Py3status:
+    # Available Configuration Parameters
+    cache_timeout = 10
+    check_pid = fallback
+    format = "VPN: {name}"
+    pidfile = '/sys/class/net/vpn0/dev_id'
+
+    def __init__(self):
+        # Create handler thread
+        self.t = Thread(target=self._start_loop)
+        self.t.daemon = True
+        self.thread_started = False
+
+    def _start_thread(self):
+        self.loop = GLib.MainLoop()
+        self.bus = dbus.SystemBus(self.loop)
+        self.bus.add_signal_receiver(self._vpn_signal_handler,
+                                     dbus_interface="org.freedesktop.NetworkManager.VPN.Connection",
+                                     signal_name="PropertiesChanged")
+        # Start handler thread
+        self.t.start()
+        self.thread_started = True
```

Comment:
To me it feels like you are trying to define the thread and start it in the same function.  If you have the 'thread' define the loop and start it separate from the `self.t.start()` stuff it might help.